scripts/publish-image.sh

```diff
@@ -19,6 +20,11 @@ exists=$(echo "$version_list" | jq --arg t "${CTR_TAG}" '.tags | index($t)')
 
 if [[ $exists == null ]]
 then
-    make docker-build-"$IMAGE_TAG"
-    docker push "$IMAGE_REPO/$IMAGE_TAG:${CTR_TAG}" || { echo "Error pushing images to container registry $(CTR_REGISTRY)/$(NAME):$(CTR_TAG)"; exit 1; }
+    if [[ $OS == "linux" ]]
+    then
+        make docker-build-"$IMAGE_TAG"
+        docker push "$IMAGE_REPO/$IMAGE_TAG:${CTR_TAG}" || { echo "Error pushing images to container registry $(CTR_REGISTRY)/$(NAME):$(CTR_TAG)"; exit 1; }
+    else
+        make ARGS=--push "docker-build-$(NAME)"
+    fi
```

Comment:
Is this missing a `docker push` for the non-linux branch of this condition?

Leaving this mostly as a note to myself for later, but maybe this script should assume the images are already built, then it can basically just do, "Given an image, check if it exists. If not, push." That should be platform-agnostic.

Reply:
The way it works is because we use buildx we need to push when we build because we cant load the image

Reply:
So added an optional args `--push` that is applied to push the image

Reply:
Ah, I misinterpreted the help for that flag. I thought it meant it just built the image in a way that it could be pushed to a registry without actually pushing it. Here I see it does actually push it: https://docs.docker.com/engine/reference/commandline/buildx_build/#push

Reply:
I will add a comment on the command to make sure it is clear for the next person